stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java

```diff
@@ -266,27 +264,36 @@ public void stop() {
     } catch (IOException e) {}
   }
 
-  private void bindToSocket(String address) {
-    try {
-      int retries = MAX_BIND_RETRIES;
-      while (retries > 0) {
-        retries--;
-        try {
-          if (LogUtil.isLoggable(Log.DEBUG)) {
-            LogUtil.d("Binding server to " + address);
-          }
-          mServerSocket = new LocalServerSocket(address);
-          LogUtil.i("Listening on @" + address);
-          return;
-        } catch (BindException be) {
-          LogUtil.w(be, "Binding error, sleep 1 second ...");
-          if (retries == 0)
-            throw be;
-          Thread.sleep(TIME_BETWEEN_BIND_RETRIES_MS);
+  @Nonnull
+  private static LocalServerSocket bindToSocket(String address) throws IOException {
+    int retries = MAX_BIND_RETRIES;
+    IOException firstException = null;
+    do {
+      try {
+        if (LogUtil.isLoggable(Log.DEBUG)) {
+          LogUtil.d("Trying to bind to @" + address);
+        }
+        return new LocalServerSocket(address);
+      } catch (BindException be) {
+        LogUtil.w(be, "Binding error, sleep 1 second ...");
+        if (firstException == null) {
+          firstException = be;
         }
+        sleepUninterruptibly(TIME_BETWEEN_BIND_RETRIES_MS);
+      }
+    } while (retries-- > 0);
+
+    throw firstException;
+  }
+
+  private static void sleepUninterruptibly(long millis) {
+    while (true) {
+      try {
+        Thread.sleep(millis);
```

Comment:
You should save state here. If you keep getting interrupted you will never terminate this loop.